precommit.py

```diff
@@ -126,44 +127,55 @@ def main() -> int:
 
     overwrite = bool(args.overwrite)
 
-    py_dir = pathlib.Path(__file__).parent
-
-    # yapf: disable
-    pths = sorted(
-        list(py_dir.glob("*.py")) +
-        list((py_dir / 'tests').glob("*.py")))
-    # yapf: enable
+    repo_root = pathlib.Path(__file__).parent
 
     print("Building the C++ module...")
-    success = build_and_install_module(py_dir)
+    success = build_and_install_module(repo_root)
     if not success:
         return 1
+
+    print("Successfully built the C++ module.")
+
+    # yapf: disable
+    print("YAPF'ing...")
+    if overwrite:
+        subprocess.check_call([
+            "yapf", "--in-place", "--style=style.yapf", "--recursive", "tests",
+            "numenc", "setup.py", "precommit.py", "bin/pynumenc"
+        ], cwd=repo_root.as_posix())
+    else:
+        subprocess.check_call([
+            "yapf", "--diff", "--style=style.yapf", "--recursive", "tests",
+            "numenc", "setup.py", "precommit.py", "bin/pynumenc"
+        ], cwd=repo_root.as_posix())
+
+    print("Mypy'ing...")
+    subprocess.check_call(["mypy", "tests", "bin/pynumenc"],
+                          cwd=repo_root.as_posix())
+
+    print("Isort'ing...")
+    if overwrite:
+        subprocess.check_call([
+            "isort", "--recursive", "tests",
+            "bin/pynumenc"], cwd=repo_root.as_posix())
     else:
-        print("Succesfully built the C++ module.")
-
-    print("There are {} file(s) that need to be individually "
-          "checked...".format(len(pths)))
-
-    futures_paths = [
-    ]  # type: List[Tuple[concurrent.futures.Future, pathlib.Path]]
-    with concurrent.futures.ThreadPoolExecutor() as executor:
-        for pth in pths:
-            future = executor.submit(
-                fn=check, path=pth, py_dir=py_dir, overwrite=overwrite)
-            futures_paths.append((future, pth))
-
-        for future, pth in futures_paths:
-            report = future.result()
-            if report is None:
-                print("Passed all checks: {}".format(pth))
-            else:
-                print(
-                    "One or more checks failed for {}:\n{}".format(pth, report))
-                success = False
+        subprocess.check_call([
+            "isort", "--check-only", "--recursive", "tests",
+            "bin/pynumenc"], cwd=repo_root.as_posix())
+
+    print("Pylint'ing...")
+    subprocess.check_call(
+        ["pylint", "--rcfile=pylint.rc", "tests", "bin/pynumenc"],
+        cwd=repo_root.as_posix())
+
+    print("Pydocstyle'ing...")
+    subprocess.check_call(["pydocstyle", "bin/pynumenc"],
+                          cwd=repo_root.as_posix())
+    # yapf: enable
 
     print("Doctesting...")
     subprocess.check_call(
-        ["python3", "-m", "doctest", (py_dir / "README.rst").as_posix()])
+        ["python3", "-m", "doctest", (repo_root / "README.rst").as_posix()])
```

Comment:
please also add all the modules -- doctests can also appear in the docstrings of the modules/functions/classes.